ulam-vs-random.py

```diff
@@ -1,19 +1,30 @@
 #!/usr/bin/env python3
 import random
 
-
+"""Seives Algo to find prime numbers within n . More Time Efficient. """
 def get_primes(n):
-    """Get all primes less than or equal to n."""
-    if n < 2:
-        return []
-    primes = [2]
-    for i in range(3, n+1):
-        for p in primes:
-            if i % p == 0:
-                break
-            if p > int(i ** 0.5):
-                primes.append(i)
-                break
+
+    primes = []
+    
+    if(n<2):
+        return primes;
+
+
+    prime = [True for i in range(n+1)]
+ 
+    i = 2
+    while (i * i <= n):
```

Comment:
I don't think any implementation of Eratosthenes' sieve writes this loop in this manner because multiplying `i` with itself in every iteration is inefficient. The square-root of `n` should be computed once and `i` should be compared against that.

Further this should be written as a for-loop to make the code more Pythonic (idiomatic). Something like:

```
    sqrt_n = n ** 0.5
    for i in range(2, int(sqrt_n) + 1):
```